### DISCUSSION TITLE
HasAttributes trait has a method to fetch only attributes you want but doesn't have a method to fetch all attributes except given on args

### DISCUSSION THREAD

**Discussion by pmiczewski** [laravel/framework#46491] [Ideas]
Has anyone ever wanted to get every attribute from a model except the ones you give on args? Of course we can achieve that in many different ways but is it viable to have an except method inside HasAttributes trait?

Somehow like
```
   public function except($attributes)
    {
        $attributes = is_array($attributes) ? $attributes : func_get_args();

        return Arr::except($this->attributes, $attributes);
    }
```
or because mutators and relations
```
 public function except($attributes)
    {
        // get all attributes from the model
        // remove attributes from function args
        // call only method with attributes that you want to fetch and return
    }
```
Maybe that's too much but here's an idea

**Comment by Angel5a** [comment]
This will greatly affect performance. It's easy to chain `getAttributes()` with `Arr::except()`, but we want except and only methods to be complementary. So we need to get the list of all possible "attributes" that can be accessed by the `only()` method. And there are attributes, relations, getters in old and new style - a bunch of runtime calls to determine them all.
